### PR TITLE
Add Contributing Information and PR Template

### PR DESCRIPTION
**Pull request by shubhanshu02**
This PR proposes the following changes:

1. Add Contributing Guidelines to the repository.
2. Update Readme with relevant repository information.
3. Introduce a Pull Request template.
4. Format the existing code (last unformatted commit).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<!-- A clear and concise description of what the pull request does. -->
+
+**Checklist:**
+
+<!-- Replace  the [ ] with [x] to check the boxes -->
+
+-   [ ] I have read the contribution guidelines in the repository.
+-   [ ] My pull request has a descriptive title (not a vague title like "Update package.json").
+-   [ ] My code follows the established code style of the repository.
+-   [ ] I have tried running the project locally and verified that there are no visible errors.
```
Comment (king-11) [inline .github/pull_request_template.md:18]: We can add something like my branch is in sync with main branch too rest LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+4. Before committing changes, you should verify if the issue is fixed and also run the following command to format the code.
+
+```bash
+npm run format
+```
+
+5. Commit your changes. Try to follow the [best practices][best_practices] while committing changes.
+
+```bash
+git commit
```
Comment (king-11) [inline CONTRIBUTING.md:48]: `git commit -s` can be introduced here as we are telling them to use `git commit`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# Then write the commit message and save changes
+```
+
+6. Push the changes to your fork.
+
+```bash
+git push origin <branch name>
```
Comment (king-11) [inline CONTRIBUTING.md:55]: this I guess should be `git push origin -u <branch name>

File: README.md [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
+The home for hackathons organized under COPS IIT (BHU), built on Next.js.
 
 ## Getting Started
 
-First, run the development server:
+For setting up the development environment, follow the steps given below.
+
+1. For making changes to the project, fork this repository using the green button on this page.
+
+2. Clone your fork or this repository, as applicable.
```
Comment (king-11) [inline README.md:12]: this repository 🤔 for what purpose?

File: CONTRIBUTING.md (hunk 2 above)
Comment (shubhanshu02) [reply CONTRIBUTING.md:48]: Is sign off necessary? :thinking:

File: README.md (hunk 4 above)
Comment (shubhanshu02) [reply README.md:12]: Maybe just for trying since this is in general, not just for contributing

File: CONTRIBUTING.md (hunk 2 above)
Comment (king-11) [reply CONTRIBUTING.md:48]: adds credibility I guess can be followed as a good practice

File: .github/pull_request_template.md (hunk 1 above)
Comment (shubhanshu02) [reply .github/pull_request_template.md:18]: How about "My branch is in sync with the branch this PR targets"?

File: CONTRIBUTING.md (hunk 2 above)
Comment (shubhanshu02) [reply CONTRIBUTING.md:48]: Sure then

File: README.md (hunk 4 above)
Comment (king-11) [reply README.md:12]: cool

File: .github/pull_request_template.md (hunk 1 above)
Comment (king-11) [reply .github/pull_request_template.md:18]: All will target main right now 🤔 we can have something simple like "My branch is up-to-date with upstream branch"

Comment (shubhanshu02) [reply .github/pull_request_template.md:18]: Yeah right!

**Pull request merged**